### PR TITLE
refactored to create a common window title & added title for electron

### PR DESCRIPTION
**Pull request by devaradhanm**
#### Description of changes

This PR refactors WindowTitle to support adding actionable icons.
Also, adds title to automated check view title bar.
Using flex to simplify styling
moved padding inside rather than on container. This way, the icons will have more hover space

Moving Window-title outside device-connect-view will be in a later pr

image

image

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/electron/views/device-connect-view/components/window-title.scss [hunk 1]
```diff
@@ -4,7 +4,23 @@
 @import '../../../../common/styles/fonts.scss';
 
 .window-title {
-    padding: 6px 9px;
+    width: 100%;
+    height: 32px;
+    display: flex;
+    align-items: center;
+
```
Comment (jalkire) [inline src/electron/views/device-connect-view/components/window-title.scss]:
```suggestion
    -webkit-app-region: drag;
```

This should already have been in `title-bar` but was not. Suggesting adding it here, but doesn't need to be in this PR.

File: src/electron/views/device-connect-view/components/window-title.scss [hunk 2]
```diff
@@ -4,7 +4,23 @@
...
+    .title-container {
+        padding: 6px 9px;
+        display: flex;
+    }
+
+    .actionable-icons-container {
+        margin-left: auto;
```
Comment (jalkire) [inline src/electron/views/device-connect-view/components/window-title.scss]:
```suggestion
        margin-left: auto;
        -webkit-app-region: no-drag;
```

If you take the other suggestion, this should accompany it to make sure the buttons still function.

File: src/electron/views/device-connect-view/components/window-title.tsx [hunk 3]
```diff
@@ -2,20 +2,31 @@
...
+            <div className={titleContainer}>
                 {props.children}
                 <h1 className={headerText}>{props.title}</h1>
             </div>
+            {getIconsContainer(props.actionableIcons)}
         </header>
     );
 });
+
+function getIconsContainer(icons?: JSX.Element[]): JSX.Element {
```
Comment (smoralesd) [inline src/electron/views/device-connect-view/components/window-title.tsx]: why outside the `NamedFC`?

File: src/electron/views/device-connect-view/components/window-title.tsx [hunk 4]
```diff
@@ -2,20 +2,31 @@
...
+    if (icons != null && icons.length > 0) {
```
Comment (smoralesd) [inline src/electron/views/device-connect-view/components/window-title.tsx]: checking for null and length sounds like a job for `lodash.isEmpty`

File: src/electron/views/device-connect-view/components/window-title.tsx (hunk 3 above)
Comment (devaradhanm) [reply src/electron/views/device-connect-view/components/window-title.tsx]: helper function.

Comment (smoralesd) [reply src/electron/views/device-connect-view/components/window-title.tsx]: it's looks very specific to the component itself... plus, it's not being used anywhere else... but up to you

**Pull request merged**